zypp/target/RpmPostTransCollector.cc

```diff
@@ -79,53 +81,83 @@ namespace zypp
 	}
 
 	/** Execute te remembered scripts. */
-	void executeScripts()
-	{
-	  if ( _scripts.empty() )
-	    return;
+  void executeScripts()
+  {
+    if ( _scripts.empty() )
+      return;
 
-	  HistoryLog historylog;
+    HistoryLog historylog;
 
-	  Pathname noRootScriptDir( ZConfig::instance().update_scriptsPath() / tmpDir().basename() );
+    Pathname noRootScriptDir( ZConfig::instance().update_scriptsPath() / tmpDir().basename() );
 
-	  for ( const auto & script : _scripts )
-	  {
-	    MIL << "EXECUTE posttrans: " << script << endl;
-            ExternalProgram prog( (noRootScriptDir/script).asString(), ExternalProgram::Stderr_To_Stdout, false, -1, true, _root );
-
-	    str::Str collect;
-	    for( std::string line = prog.receiveLine(); ! line.empty(); line = prog.receiveLine() )
-	    {
-	      DBG << line;
-	      collect << "    " << line;
-	    }
-	    int ret = prog.close();
-	    const std::string & scriptmsg( collect );
-
-	    if ( ret != 0 || ! scriptmsg.empty() )
-	    {
-	      const std::string & pkgident( script.substr( 0, script.size()-6 ) );	// strip tmp file suffix
-
-	      if ( ! scriptmsg.empty() )
-	      {
-		str::Str msg;
-		msg << "Output of " << pkgident << " %posttrans script:\n" << scriptmsg;
-		historylog.comment( msg, true /*timestamp*/);
-		JobReport::info( msg );
-	      }
-
-	      if ( ret != 0 )
-	      {
-		str::Str msg;
-		msg << pkgident << " %posttrans script failed (returned " << ret << ")";
-		WAR << msg << endl;
-		historylog.comment( msg, true /*timestamp*/);
-		JobReport::warning( msg );
-	      }
-	    }
-	  }
-	  _scripts.clear();
-	}
+    ProgressData scriptProgress( static_cast<long long>(_scripts.size()) );
+    callback::SendReport<ProgressReport> report;
+    scriptProgress.sendTo( ProgressReportAdaptor( ProgressData::ReceiverFnc(), report ) );
+
+    bool firstScript = true;
+    for ( const auto &script : _scripts )
+    {
+      const std::string & pkgident( script.substr( 0, script.size()-6 ) );	// strip tmp file suffix
+
+      scriptProgress.name( str::FormatNAC(_("Executing posttrans script '%1%'")) % pkgident );
```

Comment:
As I'm about to remove `FormatNAC`, better change it to `Format`, so it will rebase on master when merging.